torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1292,6 +1292,31 @@ int computeDeltaMS(
       .count();
 }
 
+std::string ProcessGroupNCCL::getNCCLWatchdogTimeoutErrorMsg(
+    const std::string& extraMsg) {
+  return c10::str(
+      logPrefix(),
+      "Received a dump signal due to a collective timeout from ",
+      extraMsg,
+      " and we will try our best to dump the debug info. ",
+      "Last enqueued NCCL work: ",
+      pgStatus_->lastEnqueuedSeq,
+      ", last completed NCCL work: ",
+      pgStatus_->lastCompletedSeq,
+      ".",
+      "This is most likely caused by incorrect usages of collectives, e.g., wrong ",
+      "sizes used across ranks, the order of collectives is not same for all ranks ",
+      "or the scheduled collective, for some reason, didn't run. Additionally, ",
+      "this can be caused by GIL deadlock or other reasons such as network errors or ",
+      "bugs in the communications library (e.g. NCCL), etc. ");
+}
+
+std::string ProcessGroupNCCL::getNCCLWatchdogTimeoutExitMsg() {
+  return c10::str(
+      logPrefix(),
+      "Exit processGroupNCCL with debug info dump after a collective timeout.");
```

Comment:
logPrefix already says ProcesGroupNCCL rank blah, right?

how about 

"Terminating the process after attempting to dump debug info, due to collective timeout or exception"

Reply:
It only mentions about the rank but no mentioning of `ProcesGroupNCCL `

Reply:
I kept the first two words here.